nixos/modules/services/networking/teamspeak3.nix

```diff
@@ -96,34 +95,34 @@ in
 
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    users.extraUsers.teamspeak =
-      { name = "teamspeak";
+  config = mkMerge [
+    (mkIf cfg.enable {
+      users.users.teamspeak = {
+        name = "teamspeak";
```

Comment:
If you're doing cleanup, the `name` is redundant here, it defaults to the `users` attr key.